test/e2e/flowaggregator_test.go

```diff
@@ -22,94 +22,85 @@ import (
 AntreaProxy enabled (Intra-Node): Flow record information is complete for source and destination along with K8s service info such as destinationClusterIP, destinationServicePort, destinationServicePortName etc.
 AntreaProxy enabled (Inter-Node): Flow record from destination Node is ignored, so only flow record from the source Node has its K8s info like in Inter-Node case along with K8s Service info such as destinationClusterIP, destinationServicePort, destinationServicePortName etc.
 */
 
-func TestFlowExporter(t *testing.T) {
-	// Should I add skipBenchmark as this runs iperf?
+const (
+	ingressNetworkPolicyName   = "test-flow-aggregator-networkpolicy-ingress"
+	egressNetworkPolicyName    = "test-flow-aggregator-networkpolicy-egress"
+	collectorCheckDelay        = 1 * time.Second
+	expectedNumTemplateRecords = 1
+	// Single iperf resulting in two connections with separate ports. Suspecting second flow to be control flow to exchange
+	// stats info. As 5s is export interval and iperf traffic runs for 10s, we expect about 4 records exporting to the flow aggregator.
+	// Since flow aggregator will aggregate records based on 5-tuple connection key, so we expect 2 records.
+	expectedNumDataRecords = 2
+)
+
+func TestFlowAggregator(t *testing.T) {
+	// TODO: remove this limitation after flow aggregator supports IPv6
+	skipIfIPv6Cluster(t)
+	skipIfNotIPv4Cluster(t)
 	data, err, isIPv6 := setupTestWithIPFIXCollector(t)
 	if err != nil {
 		t.Fatalf("Error when setting up test: %v", err)
 	}
+	defer teardownFlowAggregator(t, data)
 	defer teardownTest(t, data)
 
-	if err := data.createPodOnNode("perftest-a", masterNodeName(), perftoolImage, nil, nil, nil, nil, false, nil); err != nil {
-		t.Errorf("Error when creating the perftest client Pod: %v", err)
-	}
-	podAIP, err := data.podWaitForIPs(defaultTimeout, "perftest-a", testNamespace)
-	if err != nil {
-		t.Errorf("Error when waiting for the perftest client Pod: %v", err)
-	}
-
-	svcB, err := data.createService("perftest-b", iperfPort, iperfPort, map[string]string{"antrea-e2e": "perftest-b"}, false, v1.ServiceTypeClusterIP)
-	if err != nil {
-		t.Errorf("Error when creating perftest service: %v", err)
-	}
-
-	if err := data.createPodOnNode("perftest-b", masterNodeName(), perftoolImage, nil, nil, nil, []v1.ContainerPort{{Protocol: v1.ProtocolTCP, ContainerPort: iperfPort}}, false, nil); err != nil {
-		t.Errorf("Error when creating the perftest server Pod: %v", err)
-	}
-	podBIP, err := data.podWaitForIPs(defaultTimeout, "perftest-b", testNamespace)
-	if err != nil {
-		t.Errorf("Error when getting the perftest server Pod's IP: %v", err)
-	}
-
-	svcC, err := data.createService("perftest-c", iperfPort, iperfPort, map[string]string{"antrea-e2e": "perftest-c"}, false, v1.ServiceTypeClusterIP)
-	if err != nil {
-		t.Errorf("Error when creating perftest service: %v", err)
-	}
+	// Flow exporter can't connect with flow aggregator service if proxy disabled.
+	skipIfProxyDisabled(t, data)
```

Comment:
That doesn't sound right to me, could you elaborate?

Ability to connect to the flow aggregator should be independent of whether AntreaProxy is enabled.

Reply:
Yes you're right, only LocalServiceAccess and RemoteServiceAccess testcases neet to be skipped.